### DISCUSSION TITLE
Unable to reload inside an event loop using 'create_task'

### DISCUSSION THREAD

**Discussion by Corfucinas** [Kludex/uvicorn#1843] [Potential Issue] [answered]
Kindly analyze the following code snippet and clarify if this is a potential problem or the code has some flaws, currently it is not possible to load inside a running event loop

```python
# built-in
import asyncio

# external
import uvicorn
from fastapi import FastAPI


def main():
    event_loop = asyncio.new_event_loop()
    app_fastapi = FastAPI()
    server = uvicorn.Server(
        config=uvicorn.Config(
            app_fastapi, # also passing the module as main:app_fastapi and moving this to global does not work
            reload=True,
            loop="uvloop",
        ),
    )

    event_loop.create_task(server.serve())
    event_loop.run_forever()


if __name__ == "__main__":
    main()
```

The following is another snipper of code that behaves the same way as above

```python
# built-in
import asyncio

# external
import uvicorn
from fastapi import FastAPI


async def main():
    app_fastapi = FastAPI()
    server = uvicorn.Server(
        config=uvicorn.Config(
            app_fastapi,   # also passing the module as main:app_fastapi and moving this to global does not work
            reload=True,
            loop="uvloop",
        ),
    )

    api = asyncio.create_task(server.serve())

    await asyncio.wait([api])


if __name__ == "__main__":
    asyncio.run(main())

```

Once running, reload will not work upon file changes.

**Comment by Kludex** [comment, marked as answer]
That's correct. The `reload` only works when using `uvicorn.run`.

https://github.com/encode/uvicorn/blob/23b9f05a389a476aee151698cede171d915f6c53/uvicorn/main.py#L453-L574

**Reply by Corfucinas** [reply]
Is it possible to add this to the documentation? Do you want me to open an issue? PR?

**Comment by zakkg3** [comment]
Is there a workaround other than implement my own reloader?
My use case is another async app running (rocketry) so i have to run programatically and inside an asyncio event loop to make both of them work together.
https://rocketry.readthedocs.io/en/stable/cookbook/fastapi.html?highlight=fastapi
